src/merkle/merkle_tree.rs

```diff
@@ -67,18 +66,17 @@ impl MerkleTree {
     /// Returns an error if:
     /// * The specified depth is greater than the depth of the tree.
     /// * The specified index not valid for the specified depth.
-    pub fn get_node(&self, depth: u32, index: u64) -> Result<Word, MerkleError> {
-        if depth == 0 {
-            return Err(MerkleError::DepthTooSmall(depth));
-        } else if depth > self.depth() {
-            return Err(MerkleError::DepthTooBig(depth));
-        }
-        if index >= 2u64.pow(depth) {
-            return Err(MerkleError::InvalidIndex(depth, index));
+    pub fn get_node(&self, node: NodeIndex) -> Result<Word, MerkleError> {
```

Comment:
nit: should we change parameter name from `node` to `index`?

Reply:
Done!